### PR TITLE
make GoogleIdTokenAuth package-private

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
Make internal component private.

## Motivation and Context
Internals should not be public.

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-client/src/main/java/com/spotify/styx/client/GoogleIdTokenAuth.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class GoogleIdTokenAuth {
```
Comment (honnix) [inline styx-client/src/main/java/com/spotify/styx/client/GoogleIdTokenAuth.java]: Is it possible we have better test for this file? Too much code without unit test. :(

Comment (honnix) [reply styx-client/src/main/java/com/spotify/styx/client/GoogleIdTokenAuth.java]: nvm. this is a PR on top of the other one.

**Comment by honnix** [comment]
I think it is OK we apply this refactoring as part of the original PR.

**Comment by danielnorberg** [comment]
Doing file rename changes in the same PR as other changes makes it really hard to review the changes because github displays the file as completely new instead of a diff.

**Comment by honnix** [comment]
This looks very different than it was yesterday. GitHub showed me a huge diff with a new file and a deleted one. That was why it felt no difference than doing renaming in a same PR.

**Pull request merged**